src/content/docs/en/guides/cms/kontent-ai.mdx

```diff
@@ -0,0 +1,472 @@
+
+<FileTree title="Project Structure">
+- src/
+  - lib/
+    - **kontent.ts**
+- **.env**
+- astro.config.mjs
+- package.json
+</FileTree>
+
+### Fetching data
+
+The `DeliveryClient` is now available to all components. To fetch content, use the `DeliveryClient` and fluent API syntax to define your desired items. This example shows a basic fetch of blog posts and renders their titles in a list:
+
+```astro
+---
+import { deliveryClient } from "../lib/kontent";
+
+const blogPosts = await deliveryClient
+    .items()
+    .type("blogPost")
+    .toPromise()
+---
+<ul>
+  {blogPosts.data.items.map(blogPost => (
+    <li>{blogPost.elements.title.value}</li>
+  ))}
+</ul>
+```
+
+You can find more querying options in the [Kontent.ai documentation](https://kontent.ai/learn/develop/hello-world/get-content/javascript).
+
+## Making a blog with Astro and Kontent.ai
+
+With the setup above, you are now able to create a blog that uses Kontent.ai as the source of content.
+
+### Prerequisites
+
+1. **Kontent.ai project** - For this tutorial, we recommend using a blank project. If you already have some content types in your content model, you may use them, but you will need to modify the code snippets to match your content model.
+2. **Astro project configured for content fetching from Kontent.ai** - see above for more details on how to set up an Astro project with Kontent.ai
+
+### Setting up content model
+
+In Kontent.ai, navigate to **Content model** and create a new content type with the following fields and values:
+
+* **Name:** Blog Post
+* Elements:
+	* Text field
+		* **Name:** Title
+		* **Element Required:** yes
+	* Rich text field
+		* **Name:** Teaser
+		* **Element Required:** yes
+		* **Allowed in this element:** only check Text
+	* Rich text field
+		* **Name:** Content
+		* **Element Required:** yes
+	* Date & time field
+		* **Name:** Date
+	* URL slug field
+		* **Name:** URL slug
+		* **Element Required:** yes
+		* **Auto-generate from:** select "Title"
+
+Then, click on **Save Changes**.
+
+### Creating content
+
+Now, navigate to **Content & assets** tab and create a new content item of type **Blog Post**. Fill the fields using these values:
+
+* **Content item name:** Astro is amazing
+* **Title:** Astro is amazing
+* **Teaser:** Astro is an all-in-one framework for building fast websites faster.
+* **Content:** You can use JavaScript to implement the website functionality, but no client bundle is necessary.
+* **Date & time:** select today
+* **URL slug:** astro-is-amazing
+
+When you're finished, publish the blog post using the **Publish** button at the top.
+
+*Note: Feel free to create as many blog posts as you like before moving to the next step.* 
+
+### Generating content model in TypeScript
+
+Next, we'll generate TypeScript types out of your content model.
+
+*Note: This step is optional but provides a much better developer experience and allows you to discover potential problems at build time rather than at runtime.*
+
```

Comment:
This might be better using a "real" note block,

```suggestion

:::note[Note]
This step is optional but provides a much better developer experience and allows you to discover potential problems at build time rather than at runtime.
:::

```